spec/integration/grid_fs_bucket_spec.rb

```diff
@@ -30,7 +33,7 @@
 
     context 'in binary encoding' do
       let(:data_to_write) do
-        data.force_encoding('binary').freeze
+        data.dup.force_encoding('binary').freeze
```

Comment:
this should likely be in a standalone commit so the purpose of the change is clearer (it's not just adding magic comments)